integration/update_test.go

```diff
@@ -466,6 +466,58 @@ func TestUpdateSet(t *testing.T) {
 	}
 }
 
+func TestUpdateTimestamp(t *testing.T) {
+	t.Parallel()
+
+	// store the current timestamp with $currentDate operator;
+	t.Run("currentDateReadBack", func(t *testing.T) {
+		maxDifference := time.Duration(2 * time.Second)
+		now := time.Now()
+		nowTimestamp := primitive.Timestamp{T: uint32(now.Unix()), I: uint32(0)}
+		id := "string-empty"
+
+		update := bson.D{{"$currentDate", bson.D{{"value", bson.D{{"$type", "timestamp"}}}}}}
+		stat := &mongo.UpdateResult{
+			MatchedCount:  1,
+			ModifiedCount: 1,
+			UpsertedCount: 0,
+		}
+		path := types.NewPathFromString("value")
+		result := bson.D{{"_id", id}, {"value", nowTimestamp}}
+
+		ctx, collection := setup(t, shareddata.Scalars, shareddata.Composites)
+
+		res, err := collection.UpdateOne(ctx, bson.D{{"_id", id}}, update)
+		require.NoError(t, err)
+		require.Equal(t, stat, res)
+
+		// read it, check that it is close to the current time;
+		var actualB bson.D
+		err = collection.FindOne(ctx, bson.D{{"_id", id}}).Decode(&actualB)
+		require.NoError(t, err)
+
+		expected := ConvertDocument(t, result)
+		actualX := ConvertDocument(t, actualB)
```

Comment:
Those names `actualX`, and `actualB` is really confusing. Maybe there is a way to make them more meaningful?